neptune/new/handler.py

```diff
@@ -291,6 +316,7 @@ def pop(self, path: str = None, wait: bool = False) -> None:
         "fetch_files_list",
     ]
 
+    @check_protected_paths
```

Comment:
This (and a few others) won't work, I'm afraid -- you'll never reach the annotated version due to magic `__getattribute__` + `DOCSTRING_ATTRIBUTES`.

Simplest fix I can think of would be to add `super().__getattribute__(attribute_name)` to `__getattribute__` and check whether the (unused) method is wrapped in `check_protected_paths`. You're welcome to thinking up something nicer ;)

Reply:
Yes, ofc it won't work...